pkg/k8shandler/status.go

```diff
@@ -261,147 +264,183 @@ func (er *ElasticsearchRequest) updateNodeConditions(status *api.ElasticsearchSt
-					podCondition.Type = v1.PodReasonUnschedulable
-					podCondition.Status = v1.ConditionTrue
-					updatePodUnschedulableCondition(node, podCondition)
-					isUnschedulable = true
-				}
-			}
+		matchingLabels := map[string]string{
+			"component":    "elasticsearch",
+			"cluster-name": cluster.GetName(),
+			"node-name":    nodeName,
+		}
 
-			if isUnschedulable {
-				continue
-			}
-			updatePodUnschedulableCondition(node, v1.PodCondition{
-				Status: v1.ConditionFalse,
-			})
+		nodePodList, err := GetPodList(cluster.GetNamespace(), matchingLabels, er.client)
+		if err != nil {
+			return err
+		}
 
-			// if the pod can't be scheduled we shouldn't enter here
-			for _, containerStatus := range nodePod.Status.ContainerStatuses {
-				if containerStatus.Name == "elasticsearch" {
-					if containerStatus.State.Waiting != nil {
-						updatePodNotReadyCondition(
-							node,
-							api.ESContainerWaiting,
-							containerStatus.State.Waiting.Reason,
-							containerStatus.State.Waiting.Message,
-						)
-					} else {
-						updatePodNotReadyCondition(
-							node,
-							api.ESContainerWaiting,
-							"",
-							"",
-						)
-					}
-					if containerStatus.State.Terminated != nil {
-						updatePodNotReadyCondition(
-							node,
-							api.ESContainerTerminated,
-							containerStatus.State.Terminated.Reason,
-							containerStatus.State.Terminated.Message,
-						)
-					} else {
-						updatePodNotReadyCondition(
-							node,
-							api.ESContainerTerminated,
-							"",
-							"",
-						)
-					}
-				}
-				if containerStatus.Name == "proxy" {
-					if containerStatus.State.Waiting != nil {
-						updatePodNotReadyCondition(
-							node,
-							api.ProxyContainerWaiting,
-							containerStatus.State.Waiting.Reason,
-							containerStatus.State.Waiting.Message,
-						)
-					} else {
-						updatePodNotReadyCondition(
-							node,
-							api.ProxyContainerWaiting,
-							"",
-							"",
-						)
-					}
-					if containerStatus.State.Terminated != nil {
-						updatePodNotReadyCondition(
-							node,
-							api.ProxyContainerTerminated,
-							containerStatus.State.Terminated.Reason,
-							containerStatus.State.Terminated.Message,
-						)
-					} else {
-						updatePodNotReadyCondition(
-							node,
-							api.ProxyContainerTerminated,
-							"",
-							"",
-						)
+		er.pruneStatus(status, nodeIndex, nodeName, nodePodList.Items)
+		er.updatePodNodeConditions(nodeName, nodeStatus, nodePodList.Items, thresholdEnabled)
+	}
+
+	return nil
+}
+
+func (er *ElasticsearchRequest) pruneStatus(status *api.ElasticsearchStatus, nodeIndex int, nodeName string, pods []v1.Pod) {
+	clusterNodes := nodes[nodeMapKey(er.cluster.GetName(), er.cluster.GetNamespace())]
+	nodeStatus := status.Nodes[nodeIndex]
+	ll := er.L().WithValues("node", nodeStatus)
+
+	for _, node := range clusterNodes {
+		if nodeName == node.name() && node.isMissing() {
```

Comment:
The nesting here is too deep. Please invert the if condition here and on line 309 and use e.g.`if nodeName != node.name() || node.isMissing() { continue }`